src/controllers/user-controller.ts

```diff
@@ -1,260 +1,165 @@
 
-        await sgMail.send(data, false, async (err: any) => {
-          if (err) {
-            return res.status(500).json({
-              message: err.message,
-            })
-          }
+      const data = emailData(email, 'email address', emailTemp)
 
-          return res.status(200).json({
-            message: 'Email verification link sent. Check your email.',
+      await sgMail.send(data, false, async (err: any) => {
+        if (err) {
+          return res.status(500).json({
+            message: err.message,
           })
-        })
-      }
-
-      return res.status(200).json({
-        message:
-          "User is registered with google authentication. Can't change google account password.",
-      })
-    } else {
-      return res.status(404).json({
-        message: 'User with this email is not registered yet.',
+        }
       })
     }
+
+    await currentUser.save()
+    return res.status(200).json({
+      message: `User credentials updated successfully.${
+        email ? ' New email verification link sent.' : ''
+      }`,
+    })
   } catch (error: any) {
     return res.status(500).json({
       message: error.message,
     })
   }
 }
 
-export const changePassword = async (req: ChangePasswordReq, res: Response) => {
-  try {
-    const { authorization } = req.headers
-    const { password } = req.body
-
-    const token = authorization.trim().split(' ')[1]
-
-    const verified = jwt.verify(token, process.env.JWT_SECRET!)
-
-    if (verified) {
-      let email = jwt_decode<Email>(token).email
-
-      const existingUser = await User.findOne({ email })
-
-      if (!existingUser || !existingUser.password) {
-        return res.status(404).json({ message: `User is not registered!` })
-      }
-
-      const hashedPassword = await bcrypt.hash(password, 12)
-
-      await User.updateOne({ email }, { password: hashedPassword })
-
-      return res.status(200).json({ message: 'Password updated successfully' })
-    } else {
-      return res.status(401).json({
-        message:
-          'User is not authorized to change password. Token verification failed.',
-      })
-    }
-  } catch (error: any) {
-    return res.status(500).json({ message: error.message })
-  }
-}
-
-export const authorization = async (
-  req: RequestBody<AuthorizationReq>,
+export const getUserDetails = async (
+  req: RequestQuery<AccessToken>,
   res: Response
 ) => {
   try {
-    const { email, password } = req.body
-    const currentUser = await User.findOne({ email })
+    const { accessToken } = req.query
 
-    if (!currentUser || !currentUser.password) {
-      return res.status(404).json({ message: 'User not found' })
-    }
+    let email = jwt_decode<Email>(accessToken).email
 
-    if (!currentUser.verified) {
-      return res.status(403).json({
-        message: 'User can not log in because account is not verified yet',
-      })
+    if (!email) {
+      return res.status(401).json({ message: 'Enter valid JWT token' })
```

Comment:
This logic should be in middleware

Reply:
I created middleware for this route at src/middlewares/authorization-req.ts.